src/PhpSpec/Util/MethodAnalyser.php

```diff
@@ -44,28 +44,90 @@ public function reflectionMethodIsEmpty(\ReflectionMethod $method)
     }
 
     /**
+     * @param string $class
+     * @param string $method
+     *
+     * @return string
+     */
+    public function getRealDeclaringClass($class, $method)
+    {
+        $reflectionMethod = new \ReflectionMethod($class, $method);
+        $startLine = $reflectionMethod->getStartLine();
+        $endLine = $reflectionMethod->getEndLine();
+        $reflectionClass  = $this->getDeclaringClass($reflectionMethod, $startLine, $endLine);
+
+        return $reflectionClass->getName();
+    }
+
+    /**
      * @param \ReflectionMethod $reflectionMethod
      *
      * @return string
      */
     private function getCodeBody(\ReflectionMethod $reflectionMethod)
     {
-        $reflectionClass = $reflectionMethod->getDeclaringClass();
+        $endLine = $reflectionMethod->getEndLine();
+        $startLine = $reflectionMethod->getStartLine();
+        $reflectionClass = $this->getDeclaringClass($reflectionMethod, $startLine, $endLine);
 
-        $length = $reflectionMethod->getEndLine() - $reflectionMethod->getStartLine();
+        $length = $endLine - $startLine;
         $lines = file($reflectionClass->getFileName());
-        $code = join(PHP_EOL, array_slice($lines, $reflectionMethod->getStartLine()-1, $length+1));
+        $code = join(PHP_EOL, array_slice($lines, $startLine - 1, $length + 1));
 
         return preg_replace('/.*function[^{]+{/s', '', $code);
     }
 
     /**
+     * @param  \ReflectionMethod $reflectionMethod
+     * @param  int $methodStartLine
+     * @param  int $methodEndLine
+     *
+     * @return \ReflectionClass
+     */
+    private function getDeclaringClass(\ReflectionMethod $reflectionMethod, $methodStartLine, $methodEndLine)
```

Comment:
I clicked on the wrong line ;-)

getRealDeclaringClass is the one that seems wrong, as it's not really (always) returning a class.